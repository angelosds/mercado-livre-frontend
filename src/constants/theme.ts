export const ROOT_RELATIVE_SIZE = 16
