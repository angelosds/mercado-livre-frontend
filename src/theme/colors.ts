import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  primary: '#fe6000',
  secondary: '#3483fa',
  gray900: '#333333',
  gray700: '#666666',
  gray500: '#999999',
  gray300: '#eeeeee',
  white: '#ffffff',
};
