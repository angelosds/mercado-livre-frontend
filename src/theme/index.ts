import { DefaultTheme } from 'styled-components';

import { pixelToRem } from '../utils';

export const theme: DefaultTheme = {
  colors: {
    primary500: '#fee600',

    secondary100: '#c6dcf7',
    secondary500: '#3483fa',
    secondary700: '#2968c8',

    gray900: '#333333',
    gray700: '#333333',
    gray500: '#666666',
    gray400: '#999999',
    gray300: '#bbbbbb',
    gray200: '#dddddd',
    gray100: '#eeeeee',

    border200: '#f7f7f7',

    success: '#1ed760',

    white: '#ffffff',
  },
  fontSizes: {
    xs: pixelToRem(14),
    sm: pixelToRem(16),
    md: pixelToRem(24),
    lg: pixelToRem(28),
    xl: pixelToRem(46),
  },
  outline: {
    sm: '3px'
  },
  space: {
    sm: '4px',
    md: '8px',
    lg: '16px',
    xl: '32px',
  },
  radius: {
    sm: '4px'
  },
  breakpoints: {
    mobile: '565px'
  }
};
