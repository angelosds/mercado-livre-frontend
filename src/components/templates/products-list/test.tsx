import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme';

import ProductsListTemplate, { Props } from '.'

const defaultProps: Props = {
  breadcrumbsItems: ['item 1', 'item 2'],
  items: [
    {
      condition: 'Novo',
      description: 'Descrição',
      free_shipping: true,
      id: 'MLA902237011',
      picture: 'https://http2.mlstatic.com/D_NQ_NP_974179-MLB43818784989_102020-W.webp',
      price: 1200,
      sold_quantity: 30,
      state_name: 'São Paulo',
      title: 'Nome produto',
    },
    {
      condition: 'Novo',
      description: 'Descrição',
      free_shipping: true,
      id: 'MLA902237012',
      picture: 'https://http2.mlstatic.com/D_NQ_NP_974179-MLB43818784989_102020-W.webp',
      price: 1200,
      sold_quantity: 30,
      state_name: 'São Paulo',
      title: 'Nome produto',
    }
  ]
}

describe('ProductsListTemplate', () => {
  const build = ({ breadcrumbsItems, items }: Props) => {
    const { container } = render(<ThemeProvider theme={theme}><ProductsListTemplate breadcrumbsItems={breadcrumbsItems} items={items} /></ThemeProvider>)

    return { container }
  }

  it('should render the component', () => {
    const { container } = build(defaultProps)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render breadrumbs', () => {
    build(defaultProps)

    expect(screen.queryByTestId('breadcrumbs')).toBeTruthy()
  })

  it('should not render breadrumbs', () => {
    build({ ...defaultProps, breadcrumbsItems: null })

    expect(screen.queryByTestId('breadcrumbs')).toBeFalsy()
  })

  it('should render items lits', () => {
    build({ ...defaultProps, breadcrumbsItems: null })

    expect(screen.queryAllByTestId('item').length).toEqual(2)
  })

  it('should not render items lits', () => {
    build({ items: null, breadcrumbsItems: null })

    expect(screen.queryAllByTestId('item').length).toEqual(0)
  })
})
