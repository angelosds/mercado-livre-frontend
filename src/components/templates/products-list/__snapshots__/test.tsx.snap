// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsListTemplate should render the component 1`] = `
<div>
  <div
    data-testid="breadcrumbs"
  >
    <nav
      class="sc-bdvvtL cJoBZk"
    >
      <ul
        class="sc-gsDKAQ bAqmQn"
      >
        <li
          class="sc-dkPtRN gQhtXB"
        >
          item 1
          <div
            class="sc-hKwDye isMpKn"
          />
        </li>
        <li
          class="sc-dkPtRN gQhtXB"
        >
          <strong>
            item 2
          </strong>
        </li>
      </ul>
    </nav>
  </div>
  <div
    class="sc-eCImPb jXIUWd"
    data-testid="card-wrapper"
  >
    <div
      class="sc-crHmcD eePCkX"
    >
      <div
        data-testid="item"
      >
        <a
          href="/items/MLA902237011"
        >
          <div
            class="sc-jRQBWg sqqum"
          >
            <div
              class="sc-gKclnd iChrBO"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 180px; height: 180px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <img
                  data-nimg="fixed"
                  decoding="async"
                  src="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=384&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=384&q=75 2x"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
              </span>
            </div>
            <div
              class="sc-iCfMLu"
            >
              <div
                class="sc-furwcr hWGjpI"
              >
                <span
                  class="sc-jrQzAO cTzEAb"
                  data-testid="price"
                >
                  $ 
                  1.200
                  <span
                    class="sc-iqseJM jwtnvm"
                    data-testid="truck"
                  >
                    <test-file-stub
                      height="12"
                      width="12"
                    />
                  </span>
                </span>
                <span
                  class="sc-kDTinF jMFkxg"
                >
                  São Paulo
                </span>
              </div>
              <h1
                class="sc-pVTFL gizTlw"
              >
                Nome produto
              </h1>
            </div>
          </div>
        </a>
      </div>
      <div
        data-testid="item"
      >
        <a
          href="/items/MLA902237012"
        >
          <div
            class="sc-jRQBWg sqqum"
          >
            <div
              class="sc-gKclnd iChrBO"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 180px; height: 180px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <img
                  data-nimg="fixed"
                  decoding="async"
                  src="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=384&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=384&q=75 2x"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
              </span>
            </div>
            <div
              class="sc-iCfMLu"
            >
              <div
                class="sc-furwcr hWGjpI"
              >
                <span
                  class="sc-jrQzAO cTzEAb"
                  data-testid="price"
                >
                  $ 
                  1.200
                  <span
                    class="sc-iqseJM jwtnvm"
                    data-testid="truck"
                  >
                    <test-file-stub
                      height="12"
                      width="12"
                    />
                  </span>
                </span>
                <span
                  class="sc-kDTinF jMFkxg"
                >
                  São Paulo
                </span>
              </div>
              <h1
                class="sc-pVTFL gizTlw"
              >
                Nome produto
              </h1>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
